jgrapht-ext/src/main/java/org/jgrapht/ext/GraphMLImporter.java

```diff
@@ -273,15 +273,74 @@ public void setEdgeWeightAttributeName(String edgeWeightAttributeName)
      * GraphML-Attributes Values are read as string key-value pairs and passed
      * on to the {@link VertexProvider} and {@link EdgeProvider} respectively.
      * 
-     * @param input the input stream
+     * @param input the input reader
      * @param graph the output graph
      * @throws ImportException in case an error occurs, such as I/O or parse
      *         error
      */
+    @Override
     public void read(Reader input, Graph<V, E> graph)
         throws ImportException
     {
         try {
+            // parse
+            XMLReader xmlReader = createXMLReader();
+            GraphMLHandler handler = new GraphMLHandler();
+            xmlReader.setContentHandler(handler);
+            xmlReader.setErrorHandler(handler);
+            xmlReader.parse(new InputSource(input));
+
+            // read result
+            handler.updateGraph(graph);
+        } catch (Exception se) {
+            throw new ImportException("Failed to parse GraphML", se);
+        }
+    }
+    
```

Comment:
consider wrapping it into a InputStreamReader to reduce code duplication:  read(new InputStreamReader(input, StandardCharsets.UTF_8), graph);